pandas/core/dtypes/missing.py

```diff
@@ -394,3 +394,10 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def remove_na_arraylike(series):
+    """
+    Return series containing only true/non-NaN values, possibly empty.
+    """
```

Comment:
I know you and @jreback discussed this briefly, but I'd like to understand this:

Why are you saying "array-like" when the function only accepts `Series` (or is only meant for `Series`)?  If it isn't meant just for `Series`, we should update the documentation to reflect this.